reader.go

```diff
@@ -0,0 +1,52 @@
+package kinesumer
+
+import (
+	"io"
+
+	k "github.com/remind101/kinesumer/interface"
+)
+
+// Reader provides an io.Reader implementation that can read data from a kinesis
+// stream.
+type Reader struct {
+	// buf is used internally to buffer up received messages.
+	buf []byte
+
+	records <-chan k.Record
+}
+
+// NewReader returns a new Reader instance that reads data from records.
+func NewReader(records <-chan k.Record) *Reader {
+	return &Reader{records: records}
+}
+
+// Read implements io.Reader Read.
+func (r *Reader) Read(p []byte) (n int, err error) {
+	if len(r.buf) != 0 {
+		// Copy up to len(p) bytes from the internal buffer into p.
+		n += copy(p, r.buf[:len(p)])
+		r.buf = r.buf[len(p):]
+	}
+
+	for record := range r.records {
```

Comment:
Add a timeout here? Otherwise we'd end up blocking at the end of a stream, which seems useful but breaks convention:

> Read conventionally returns what is available instead of waiting for more.

As a consumer of this interface, I feel like I'd want to configure how long to wait for.